server_environment/models/server_env_mixin.py

```diff
@@ -0,0 +1,405 @@
+                base_fields = super()._server_env_fields
+                sftp_fields = {
+                    "sftp_server": {},
+                    "sftp_port": {'getter': "getint"},
+                    "sftp_login": {},
+                    "sftp_password": {
+                        "no_default_field": True,
+                        "compute_default": "_compute_password",
+                        "inverse_default": "_inverse_password",
+                    },
+                }
+                sftp_fields.update(base_fields)
+                return sftp_fields
+
+    * ``no_default_field`` means that no new (sparse) field need to be
+      created, it already is provided by keychain
+    * ``compute_default`` is the name of the compute method to get the default
+      value when no key is set in the configuration files.
+      ``_compute_password`` is implemented by ``keychain.backend``.
+    * ``inverse_default`` is the name of the compute method to set the default
+      value when it is editable. ``_inverse_password`` is implemented by
+      ``keychain.backend``.
+
+    """
+    _name = 'server.env.mixin'
+
+    server_env_defaults = fields.Serialized()
+
+    _server_env_getter_mapping = {
+        'integer': 'getint',
+        'float': 'getfloat',
+        'monetary': 'getfloat',
+        'boolean': 'getboolean',
+        'char': 'get',
+        'selection': 'get',
+        'text': 'get',
+    }
+
+    @property
+    def _server_env_fields(self):
+        """Dict of fields to replace by fields computed from env
+
+        To override in models. The dictionary is:
+        {'name_of_the_field': options}
+
+        Where ``options`` is a dictionary::
+
+            options = {
+                "getter": "getint",
+                "no_default_field": True,
+                "compute_default": "_compute_password",
+                "inverse_default": "_inverse_password",
+            }
+
+        * ``getter``: The configparser getter can be one of: get, getboolean,
+          getint, getfloat. The getter is automatically inferred from the
+          type of the field, so it shouldn't generally be needed to set it.
+        * ``no_default_field``: disable creation of a field for storing
+          the default value, must be used with ``compute_default`` and
+          ``inverse_default``
+        * ``compute_default``: name of a compute method to get the default
+          value when no key is present in configuration files
+        * ``inverse_default``: name of an inverse method to set the default
+          value when the value is editable
+
+        Example::
+
+            @property
+            def _server_env_fields(self):
+                base_fields = super()._server_env_fields
+                sftp_fields = {
+                    "sftp_server": {
+                        "getter": "get",
+                    },
+                    "sftp_port": {
+                        "getter": "getint",
```

Comment:
I'd remove getters where not need and ad the optional example in case you have to convert to different values